### PR TITLE
Improving telemetry error messages

### PR DESCRIPTION
**Pull request by laileni-aws**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description
- Improving telemetry error messages.

## Checklist
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanSession.kt [hunk 1]
```diff
@@ -286,7 +286,13 @@ class CodeWhispererCodeScanSession(val sessionContext: CodeScanSessionContext) {
         )
     } catch (e: Exception) {
         LOG.debug { "Create Upload URL failed: ${e.message}" }
-        throw codeScanServerException("CreateUploadUrlException: " + e.message?.let { it } ?: message("codewhisperer.codescan.run_scan_error_telemetry"))
+
+        val errorMessage = when {
+            e.message?.contains("Your account is not authorized to make this call.") == true -> "Your account is not authorized to make this call."
+            e.message?.contains("Service returned HTTP status code 407") == true -> "Service returned HTTP status code 407"
+            else -> e.message ?: message("codewhisperer.codescan.run_scan_error_telemetry")
```
Comment (ashishrp-aws) [inline plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanSession.kt:293]: are there any other exception cases we are missing here.

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanSession.kt [hunk 2]
```diff
@@ -341,7 +347,12 @@ class CodeWhispererCodeScanSession(val sessionContext: CodeScanSessionContext) {
             )
         } catch (e: Exception) {
             LOG.debug { "Creating security scan failed: ${e.message}" }
-            throw codeScanServerException("CreateCodeScanException: " + e.message?.let { it } ?: message("codewhisperer.codescan.run_scan_error_telemetry"))
+            val errorMessage = when {
+                e.message?.contains("Too many requests, please wait before trying again.") == true -> "Too many requests, please wait before trying again."
+                e.message?.contains("Improperly formed request.") == true -> "Improperly formed request."
+                else -> e.message ?: message("codewhisperer.codescan.run_scan_error_telemetry")
```
Comment (ashishrp-aws) [inline plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanSession.kt:353]: same here for any other expected exceptions from service?

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanSession.kt (hunk 1 above)
Comment (laileni-aws) [reply plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanSession.kt:293]: I found these from the Kibana. As we get more metrics we need to add if required.

File: plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanSession.kt (hunk 2 above)
Comment (laileni-aws) [reply plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanSession.kt:356]: I found these from the Kibana. As we get more metrics we need to add if required.

**Review by ashishrp-aws — APPROVED** [review]

**Pull request merged**